### DISCUSSION TITLE
Quick Question: Making a plugin for Remark

### DISCUSSION THREAD

**Discussion by samfcheng** [remarkjs/remark#1170] [Q&A] [answered]
I've created a plugin that creates these emoji blocks:
![image](https://github.com/remarkjs/remark/assets/51756184/717352f9-c5b7-4b0c-b7e9-68ebae0fdb51)
To do this, I use this code:
```js
export default function emojiBlock() {
  return function transformer(tree) {
    tree.children.forEach((node) => {
      if (node.type === 'paragraph') {
        const text = node.children[0].value
        const emoji = text.match(/^[\uD800-\uDBFF][\uDC00-\uDFFF]/g)
        const index = text.indexOf('>');
        
        if (emoji && index !== -1) {
          const colorRegex = /\[color:\s*([#\w]+|\b(?:rgb|hsl)a?\(.+?\))\]/;
          const titleRegex = /\[title:\s*([^\]]+)\]/;
          
          const restOfString = text.slice(index + 1).replace(colorRegex, '').replace(titleRegex, '');

          const color = text.match(colorRegex);
          const title = text.match(titleRegex);
          
          node.type = 'html'
          node.value = `<div class="emoji-block" style="--emoji: '${emoji[0]}'; background-color: ${color ? color[1] : '#E2E8F0'};">
            <h3>${title ? title[1] : ""}</h3>
            <p>${restOfString}</p>
          </div>`
        }
      }
    })
  }
}
```
There's probably a better way to do this, but I couldn't find anything searching through Google and the docs didn't help.
Here's the issue: Everything works fine, but if I want to include other markdown inside of the block, it just deletes that section.

e.g
`👋> [title: Hi!] Hey **there**` should have a `p` element that looks like `<p>Hey <strong>there</strong></p>`, but ends up just being `<p>Hey </p>`

How can I fix this?

**Comment by Murderlon** [comment, marked as answer]
Hi, I strongly recommend to not invent your own markdown syntax nor to work with regex if you must. Take a look at [`remark-directive`](https://github.com/remarkjs/remark-directive), which is what most of the community uses to create admonitions (the block you're trying to make).
